examples/go/tagging/gotag-fn/handler.go

```diff
@@ -0,0 +1,177 @@
+package function
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+
+	handler "github.com/openfaas-incubator/go-function-sdk"
+	"github.com/pelletier/go-toml"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vapi/tags"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+// tagInfo is part of the toml vcconfig file
+type tagInfo struct {
+	Action string
+	Urn    string
+}
+
+// VCenter is part of the toml vcconfig file
+type vCenterAuth struct {
+	Password string
+	Server   string
+	User     string
+}
+
+// Config represents the toml vcconfig file
+type vcConfig struct {
+	Tag     tagInfo
+	VCenter vCenterAuth
+}
+
+// The request from vmware-event-router.
+type dataObj struct {
+	Data nest1 `json:"data"`
+}
+type nest1 struct {
+	Vm nest2 `json:"Vm"`
+}
+type nest2 struct {
+	Vm vmInfo `json:"Vm"`
+}
+type vmInfo struct {
+	Value string `json:"Value"`
+	Type  string `json:"Type"`
+}
+
+// Handle a function invocation
+func Handle(req handler.Request) (handler.Response, error) {
+	ctx := context.Background()
+
+	// Retrieve vCenter and tag info from the config file.
+	vcc := loadTomlCfg()
+
+	// Incoming request object.
+	var mor dataObj
+	err := json.Unmarshal(req.Body, &mor)
```

Comment:
Is `Body` an `io.Closer` that needs to be closed explicitly?

Reply:
I see that in the [golang-http middleware example](https://github.com/openfaas-incubator/golang-http-template), there is a `defer r.Body.Close()` in the example where `r` is the `*http.Request`. 

However, when I looked into why we need to do a Body.Close() at all, it is so a client can signal to the server that it is done with the response, and the connection can be used by the next client request. Close() doesn't close connections--it keeps them open, but makes it available for the next request.

What that means is as a server, we don't need to close requests. 

It's possible I totally misunderstood all the explanations I read. What do you think? Here are a couple of my sources:
(1) https://grokbase.com/t/gg/golang-nuts/138p15ng41/go-nuts-http-request-body-close
In the first source, Brad Fitzpatrick wrote, "You don't need to close it for server requests. It's unfortunate in some ways that we use Request for both client and server requests."

Caleb Spare said, "The only places where the net/http documentation mentions closing req.Body
is for the client usages (e.g., Get())."

(2) https://groups.google.com/forum/#!topic/golang-nuts/K9e0tyfU2hY
Here in the second source, Jesse McNelis writes, "HTTP is a request/response protocol. You can't send a response without a request and you can't receive a new request until you've processed the previous request.
The http pkg uses keep-alive to avoid having to make a new connection for each request, but this means that it needs to clear the previous request so it can wait for a new  one and that's what r.Body.Close() does."

Reply:
I would have to dig into the specifics of how the server-side code in net/http handles closing the request body (this is a POST). I would assume, since you open the body for reading, it can still be considered good practice to close the body as soon as your logic is done, i.e. via `defer ...`. IMHO won't harm and signals that the authors thinks of preventing leakage of associated resources, e.g. file descriptor, etc.

Reply:
I don't think I open the body for reading. `Req` looks like it's just a struct, and I'm accessing the body, which is a slice of bytes. It's not the same on the client side (where the POST request is made), where the response body implements io.Writer.

Here's the function signature for handle(): 
```go
func Handle(req handler.Request) (handler.Response, error)
```

And here's the handler.Request type:
```go
type Request struct {
	Body        []byte
	Header      http.Header
	QueryString string
	Method      string
	Host        string
}
```
Source: https://github.com/openfaas-incubator/go-function-sdk/blob/70701da50a91/handler.go#L21

What does it mean to close a slice of bytes?

Reply:
@embano1 spoke about this, and there is no need to close body.